lib/blocks.php

```diff
@@ -296,3 +296,208 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array  $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
+	// Don't filter template part blocks since we filter the blocks in each template
+	// part individually.
+	if ( 'core/template-part' === $block['blockName'] ) {
+		return $block_content;
+	}
+
+	if ( isset( $block['attrs'] ) ) {
+		// Check what style features the block supports.
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+		if ( $block_type ) {
+			$supports = gutenberg_experimental_global_styles_get_supported_styles( $block_type->supports );
+		} else {
+			$supports = array();
+		}
+		// Return early if nothing is supported.
+		if ( count( $supports ) === 0 ) {
+			return $block_content;
+		}
```

Comment:
Could do

```suggestion
		if ( ! $block_type ) {
			return $block_content;
		}
		
		$supports = gutenberg_experimental_global_styles_get_supported_styles( $block_type->supports );

		// Return early if nothing is supported.
		if ( count( $supports ) === 0 ) {
			return $block_content;
		}
```

which I personally find a bit clearer (since it avoids creating the empty array just to bail right after). Not that it mattered too much tho :slightly_smiling_face:

Reply:
That works for me.  At first I was trying to avoid having 2 separate returns, but after looking at the alternative it does seem a bit more clean.  Agreed and committed 😄